openarm/damiao/__init__.py

```diff
@@ -35,532 +65,1221 @@ class MotorType(StrEnum):
-
-    MIT = 1
-    POS_VEL = 2
-    VEL = 3
-    POS_FORCE = 4
-
-
-def enable_command(slave_id: int) -> can.Message:
-    """Create a CAN message to enable a Damiao motor.
-
-    Args:
-        slave_id (int): Slave ID for the target motor.
-
-    Returns:
-        can.Message: A CAN message that, when sent, enables the motor.
+class Motor:
+    """High-level interface for controlling a Damiao motor.
 
+    This class combines encode/decode functions from encoding.py to provide
+    a convenient interface that follows the request-response pattern.
+
+    The motor uses two IDs:
+    - slave_id: For sending commands TO the motor
+    - master_id: For receiving responses FROM the motor
+
+    These IDs can be the same or different depending on motor configuration.
+
+    Reference: README.md High-Level Motor Class section lines 320-345
     """
-    return can.Message(
-        arbitration_id=slave_id,
-        data=[0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFC],
-        is_extended_id=False,
-    )
 
+    def __init__(
+        self,
+        bus: Bus,
+        *,
+        slave_id: int,
+        master_id: int,
+        motor_type: MotorType,
+    ) -> None:
+        """Initialize a Motor instance.
 
-def disable_command(slave_id: int) -> can.Message:
-    """Create a CAN message to disable a Damiao motor.
+        Args:
+            bus: CAN bus instance for message transmission and reception
+            slave_id: Motor slave ID for sending commands TO the motor
+            master_id: Motor master ID for receiving responses FROM the motor
+            motor_type: Motor type enum for automatic limit configuration
 
-    Args:
-        slave_id (int): Slave ID for the target motor.
+        Note: slave_id, master_id, and motor_type must be specified as keyword args.
 
-    Returns:
-        can.Message: A CAN message that, when sent, disables the motor.
+        Reference: DM_CAN.py Motor class with SlaveID and MasterID
 
-    """
-    return can.Message(
-        arbitration_id=slave_id,
-        data=[0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFD],
-        is_extended_id=False,
-    )
+        """
+        self._bus = bus
+        self._slave_id = slave_id  # SlaveID for sending commands
+        self._master_id = master_id  # MasterID for receiving responses
+        self._motor_type = motor_type
+        self._motor_limits = MOTOR_LIMITS[motor_type]
+
+    @property
+    def bus(self) -> Bus:
+        """Get CAN bus instance."""
+        return self._bus
+
+    @property
+    def slave_id(self) -> int:
+        """Get motor slave ID for sending commands."""
+        return self._slave_id
+
+    @property
+    def master_id(self) -> int:
+        """Get motor master ID for receiving responses."""
+        return self._master_id
+
+    @property
+    def motor_type(self) -> MotorType:
+        """Get motor type."""
+        return self._motor_type
+
+    def set_control_mode(self, mode: ControlMode) -> Coroutine[Any, Any, int]:
```

Comment:
I think it should return `ControlMode`:

```suggestion
    def set_control_mode(self, mode: ControlMode) -> Coroutine[Any, Any, ControlMode]:
```

Reply:
right!